### DISCUSSION TITLE
STM32L476RG Comparator Enable

### DISCUSSION THREAD

**Discussion by mishraEmb** [stm32duino/Arduino_Core_STM32#1752] [Q&A] [answered]
Does Arduino IDE support embedded C register level programming of any STM32 based micro board? Is any prebuild library available of Comparator using HAL/ LL supported by Arduino IDE?

**Comment by fpistm** [comment, marked as answer]
This core is based on the official HAL/LL from ST including the CMSIS. So you can do the same than using the Cube FW in Arduino.

**Reply by mishraEmb** [reply]
Thank you for the reply, and yes I had tried the same whatever function generated by STM32Cube IDE, I have implemented the same in Arduino IDE, but stuck in comparator callback function and not enterning into main loop.

**Reply by fpistm** [reply]
Think that the sketch (*.ino) is a cpp file so some function have to be declared as `extern "C"`.
Here an example of a Cube project ported to Arduino:
https://www.stm32duino.com/viewtopic.php?f=41&t=110

**Reply by mishraEmb** [reply]
Thank you for your help. I have implemented all the application functions generated via STM32Cube IDE into Arduino IDE, and it is working at some point not full functional. I have enabled comp1 and giving output of comp1 into timer input capture pin. Only one interrupt I can read rest timer stuck in callback function. Is there any other methods to write the program using register level/ embedded C/ Bare Metal approach using Arduino IDE with STM32 micro?

**Reply by fpistm** [reply]
As stated you have access to the CMSIS device definitions which contains all Register address, offset mask. up to you to use it or not like HAL/LL do.
